### PR TITLE
Update realm-core to v13.1.0

### PR DESCRIPTION
**Pull request by nielsenko**

### CODE REVIEW

**Review by desistefanova — APPROVED** [review]
There are some changes in https://github.com/realm/realm-dart/pull/974 that test if `beginWriteAsync` and `commitWriteAsync` throw the correct error. I will apply them separately after we merge this PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,20 +3,29 @@
...
+* Most of the file growth caused by version pinning is eliminated. (Core upgrade)
+* Fetching a user's profile while the user logs out would result in an assertion failure. (Core upgrade)
+* Removed the ".tmp_compaction_space" file being left over after compacting a Realm on Windows. (Core upgrade).
+* Restore fallback to full barrier when F_BARRIERSYNC is not available on Apple platforms. (Core upgrade, since v0.8.0+rc)
+* Fixed wrong assertion on query error that could result in a crash. (Core upgrade)
 
 ### Compatibility
-* Realm Studio: 12.0.0 or later.
+* Realm Studio: 13.0.0 or later.
+* Fileformat: Generates files with format v23. Reads and automatically upgrade from fileformat v5.
```
Comment (blagoev) [inline CHANGELOG.md:25]:
```suggestion
* Fileformat: Generates files with format v23. Reads and automatically upgrades from fileformat v5.
```
should this be `upgrades` vs `upgrade`

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,20 +3,29 @@
...
 
 ### Internal
-* Using Core 12.12.0.
+* Using Core 13.1.0. ([#1042](https://github.com/realm/realm-dart/pull/1042))
```
Comment (blagoev) [inline CHANGELOG.md:28]:
```suggestion
* Using Core 13.1.0.
```

File: lib/src/session.dart [hunk 3]
```diff
@@ -233,16 +228,11 @@ class SessionConnectionStateController {
...
-
     _token = realmCore.sessionRegisterConnectionStateNotifier(_session, this);
   }
 
   void _stop() {
-    if (_token == null) {
-      return;
-    }
-
-    realmCore.sessionUnregisterConnectionStateNotifier(_session, _token!);
```
Comment (blagoev) [inline lib/src/session.dart:245]: Are we sure we don't need to unregister anymore? Or perhaps, that is the new change in Core?

**Review by blagoev — APPROVED** [review]
some small suggestions

**Comment by blagoev** [comment]
>I will apply them separately after we merge this PR.
@desistefanova Should we create an issue for that?

File: CHANGELOG.md (hunk 2 above)
Comment (nielsenko) [reply CHANGELOG.md:28]: Why not specify the PR it was updated in?

File: lib/src/session.dart (hunk 3 above)
Comment (nielsenko) [reply lib/src/session.dart:245]: Yes, they changed it. Now you `realm_release` a type of notification token instead

Comment (blagoev) [reply lib/src/session.dart:245]: makes sense

File: CHANGELOG.md (hunk 2 above)
Comment (blagoev) [reply CHANGELOG.md:28]: we are not doing that in any SDK

**Pull request merged**